### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by RandyHucker**
- Added a placeholder for unvalid thumbnails.
- Fixed URL for thangs link
- Added exceptions for breaking points
- Fixed import flow

### CODE REVIEW

File: thangs_fetcher.py [hunk 1]
```diff
@@ -685,22 +701,46 @@ def get_stl_search(self, stl_path):
...
+            self.searching = False
+            self.newSearch = False
+            self.SelectionFailed = True
+            return
 
         print("Select Search Returned")
 
+        try:
+            url_filepath = urllib.parse.quote(new_Filename, safe='')
+            url_filepath = re.sub(r'-', '%2D', url_filepath)
```
Comment (craigv-physna) [inline thangs_fetcher.py:724]: Curious -- why are we replacing `-`?  IIRC it's valid in a url.

File: __init__.py [hunk 2]
```diff
@@ -528,11 +579,11 @@ def drawView(self, context):
                     row = col.row()
                     row.label(
                         text="", icon_value=icons_dict["CreativeC"].icon_id)
-                    if model[3] == "":
+                    if model[3] == "" or not str(model[3]).endswith("showLicense=true"):
```
[the remark this answers is not in the document]
Comment (randalhucker) [reply __init__.py:582]: Awesome.

File: thangs_fetcher.py (hunk 1 above)
Comment (RandyHucker) [reply thangs_fetcher.py:724]: Colin had said "The slashes and hyphens are most likely whats causing the issue", but he's looking into them right now I believe. Anything in the STL search isn't finished, just a placeholder.

File: thangs_fetcher.py [hunk 3]
```diff
@@ -425,10 +430,14 @@ def get_http_search(self):
                 self.enumItems.append(
                     (modelTitle, modelId, item["ownerUsername"], item["license"]))#, item["originalFileType"]))
 
-                print(f'Fetching {thumbnail}')
-                filePath = urllib.request.urlretrieve(thumbnail)
-
-                filepath = os.path.join(modelId, filePath[0])
+                thumbnail = thumbnail.replace("https", "http", 1)
```
[the remark this answers is not in the document]
Comment (RandyHucker) [reply thangs_fetcher.py:433]: It was, I'll look at your workaround - thanks!

File: __init__.py [hunk 4]
```diff
@@ -312,27 +336,38 @@ def execute(self, _context):
...
+        except:
+            print("Error with Loging In")
+        return
 
-        f = open(bearer_location)
-        data = json.load(f)
-        fetcher.bearer = data["bearer"]
-        thangs_api.bearer = data["bearer"]
+    def execute(self, _context):
+        print("Starting Loggin and Import")
```
Comment (craigv-physna) [inline __init__.py:366]:
```suggestion
        print("Starting Login and Import")

```

File: __init__.py [hunk 5]
```diff
@@ -312,27 +336,38 @@ def execute(self, _context):
...
+            data = json.load(f)
+            fetcher.bearer = data["bearer"]
+            thangs_api.bearer = data["bearer"]
+            print("Before Import")
+            print("Starting Download")
+            print("Import Thread Returned")
+            thangs_api.handle_download(modelIndex)
+            Model_Event(modelIndex)
+        except:
+            print("Error with Loging In")
```
Comment (craigv-physna) [inline __init__.py:362]:
```suggestion
            print("Error with Logging In")

```

File: __init__.py [hunk 6]
```diff
@@ -519,20 +583,26 @@ def drawView(self, context):
...
+                            'wm.browse_to_creator', text="%s" % model[2])
+                        props.url = thangs_config.thangs_config['url'] + "designer/" + urllib.parse.quote(str(
+                            model[2])) + "/?utm_source=blender&utm_medium=referral&utm_campaign=blender_extender"
+                        props.modelIndex = z
 
                     row = col.row()
                     row.label(
                         text="", icon_value=icons_dict["CreativeC"].icon_id)
-                    if model[3] == "":
+                    #if model[3] == "" or not str(model[3]).endswith("showLicense=true"):
```
Comment (craigv-physna) [inline __init__.py:600]:
```suggestion
```
Might as well clean this up

File: thangs_fetcher.py [hunk 7]
```diff
@@ -425,10 +430,18 @@ def get_http_search(self):
...
                     (modelTitle, modelId, item["ownerUsername"], item["license"]))#, item["originalFileType"]))
 
-                print(f'Fetching {thumbnail}')
-                filePath = urllib.request.urlretrieve(thumbnail)
+                print("-----")
+                print("-----")
+                print(self.enumItems)
 
-                filepath = os.path.join(modelId, filePath[0])
+                #thumbnail = thumbnail.replace("https", "http", 1)
```
Comment (craigv-physna) [inline thangs_fetcher.py:437]:
```suggestion
```

**Review by craigv-physna — APPROVED** [review]
Had a few small nits and cleanup things, but otherwise looks good to go

**Pull request merged**